packages/trader/src/Stores/Modules/Trading/Actions/test.ts

```diff
@@ -0,0 +1,23 @@
+import { TicksStreamResponse } from '@deriv/api-types';
+import { WS } from '@deriv/shared';
+import { TError } from 'Types';
+
+type TCallback = (data: string) => void;
+/* This action does not modify state directlly.
+ * The payload will be the callback that get's called for each tick
+ */
+let cb: TCallback;
+const ticksCallback = (response: TicksStreamResponse) => {
+    const data: string = response.error
+        ? (response.error as TError).message
+        : `${new Date(Number(response.tick?.epoch) * 1000).toUTCString()}: ${response.tick?.quote}`;
+    cb(data);
+};
```

Comment:
```suggestion
const ticksCallback = (response: TicksStreamResponse) => {
    if (response.error && typeof response.error === 'object') {
        return cb(
            'message' in response.error && typeof response.error.message === 'string' ? response.error.message : ''
        );
    }

    return cb(`${new Date(Number(response.tick?.epoch) * 1000).toUTCString()}: ${response.tick?.quote}`);
};
```
@maryia-deriv Let's not type cast, the `TError` type is useless here since there is no type check. The other way to do it is to create a type guard:

```ts
const isServerError = (error: unknown): error is TError =>
    typeof error === 'object' && error !== null && 'code' in error && 'message' in error;

const ticksCallback = (response: TicksStreamResponse) => {
    if (isServerError(response.error)) return cb(response.error.message);

    return cb(`${new Date(Number(response.tick?.epoch) * 1000).toUTCString()}: ${response.tick?.quote}`);
};
```